packages/webapp/pages/onboarding.tsx

```diff
@@ -67,27 +82,90 @@ const seo: NextSeoProps = {
+  isOnboardingV3: boolean;
+  setAuth: Dispatch<SetStateAction<AuthProps>>;
+}
+const Header = ({
+  showOnboardingPage,
+  isOnboardingV3,
+  setAuth,
+}: HeaderProps) => {
+  if (!isOnboardingV3 || !showOnboardingPage) {
+    return (
+      <Logo
+        className="py-8 px-10 w-auto laptop:w-full"
+        position={LogoPosition.Relative}
+      />
+    );
+  }
+
+  return (
+    <header className="flex justify-between px-6 mt-20 w-full h-full flew-row max-w-[75rem]">
+      <Logo
+        className="w-auto"
+        logoClassName="h-logo-big"
+        position={LogoPosition.Relative}
+      />
+
+      <span
+        className={classNames('flex items-center', 'text-theme-label-tertiary')}
+      >
+        <span className="hidden tablet:block">Already a daily.dev member?</span>
+        <Button
+          className="ml-3 btn-secondary"
+          onClick={() => {
+            setAuth({
+              isAuthenticating: true,
+              isLoginFlow: true,
+              defaultDisplay: AuthDisplay.Default,
+            });
+          }}
+        >
+          Log in
+        </Button>
+      </span>
+    </header>
+  );
+};
+
 export function OnboardPage(): ReactElement {
   const router = useRouter();
   const [showCookie, acceptCookies, updateCookieBanner] = useCookieBanner();
   const isTracked = useRef(false);
   const { user, isAuthReady } = useAuthContext();
   const [isFiltering, setIsFiltering] = useState(false);
   const [finishedOnboarding, setFinishedOnboarding] = useState(false);
-  const [auth, setAuth] = useState<AuthProps>({
-    isAuthenticating: false,
-    isLoginFlow: false,
-  });
-  const { isAuthenticating, isLoginFlow } = auth;
   const { onShouldUpdateFilters } = useOnboardingContext();
   const onboardingV2 = useFeature(feature.onboardingV2);
+  const onboardingV3 = useFeature(feature.onboardingV3);
+  const isOnboardingV3 = onboardingV3 !== OnboardingV3.Control;
   const { growthbook } = useGrowthBookContext();
   const filteringTitle = useFeature(feature.onboardingFilterTitle);
   const { onboardingIntroduction } = useThemedAsset();
   const { trackEvent } = useAnalyticsContext();
   const { alerts } = useContext(AlertContext);
   const [hasSelectTopics, setHasSelectTopics] = useState(false);
+  const [auth, setAuth] = useState<AuthProps>({
+    isAuthenticating: isOnboardingV3 && !!storage.getItem(SIGNIN_METHOD_KEY),
+    isLoginFlow: false,
+    defaultDisplay: isOnboardingV3
+      ? AuthDisplay.OnboardingSignup
+      : AuthDisplay.Default,
+  });
+  const { isAuthenticating, isLoginFlow, email, defaultDisplay } = auth;
 
+  const formRef = useRef<HTMLFormElement>();
+  const title = versionToTitle[filteringTitle];
+
+  useEffect(() => {
+    if (isOnboardingV3) {
+      setAuth({ ...auth, defaultDisplay: AuthDisplay.OnboardingSignup });
+    }
+    // Ignore the auth prop to avoid infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOnboardingV3]);
```

Comment:
You can use `setAuth(currentAuth => ({ ...currentAuth }))` to avoid breaking rules of hooks.

Reply:
Doh! Of course. I've been working too much with k8s, so I had forgotten that method 😬 
Fixed now.